juju/juju
# [JUJU-2571] Pass series in charm upgrade series hook environment

When migrating to juju 3.0, a lot of changes were made to ensure that base was in all of the API facade and state changes. Pushing series to the edge. Unfortunately, this meant that we push base into the JUJU_TARGET_SERIES, which will break a lot of charms when attempting to upgrade a machine. The solution is to send both series and base as two independent env vars. JUJU_TARGET_SERIES contains the series and JUJU_TARGET_BASE is the new base. This allows charms to migrate away from series to base in this context.

I was also considering renaming the core model for upgrade series so that it better reflects the machine status.

## Checklist


## QA steps

Ensure that there are no issues when upgrading a machine.

```sh
$ juju bootstrap lxd test
$ juju deploy ubuntu --series focal
$ juju upgrade-machine 0 prepare jammy
$ juju debug-log
```

## Bug reference

https://bugs.launchpad.net/juju/+bug/2003858
